cmd/build/v2/build.go

```diff
@@ -184,7 +184,9 @@ func (bc *OktetoBuilder) buildSvcFromDockerfile(ctx context.Context, manifest *m
 	isStackManifest := manifest.Type == model.StackType
 	buildSvcInfo := getBuildInfoWithoutVolumeMounts(manifest.Build[svcName], isStackManifest)
 
-	buildSvcInfo.AddArgs(bc.buildEnvironments)
+	if err := buildSvcInfo.ExpandBuildArgs(bc.buildEnvironments); err != nil {
```

Comment:
Before, the `AddArgs` function weren't expanding the build arguments and now, we are doing it. Why? When were the build args expanded before?

Reply:
We were doing half of the process before. We were creating `OKTETO_BUILD` env vars as args but we weren't expanding those args if the user used a custom key such as:
```yaml
build:
  args:
    FROM_IMAGE: ${OKTETO_BUILD_APP_IMAGE}
```

Reply:
Ah ok, so we were not expanding them before?

Reply:
Not expanding, we were just adding the previous args:
```yaml
build:
  args:
    FROM_IMAGE: ${OKTETO_BUILD_APP_IMAGE}
```
was converted into:
```yaml
build:
  args:
    FROM_IMAGE: # empty
    OKTETO_BUILD_APP_IMAGE: real_image_sha
```